cmake/app/boilerplate.cmake

```diff
@@ -274,6 +274,36 @@ set(KERNEL_EXE_NAME   ${KERNEL_NAME}.exe)
 set(KERNEL_STAT_NAME  ${KERNEL_NAME}.stat)
 set(KERNEL_STRIP_NAME ${KERNEL_NAME}.strip)
 
+# Try to populate USER_HOME_DIR with a directory that user
+# applications may write to. Leave it unset if unable.
+foreach(env_var
```

Comment:
Could you please let the user specify this on the CMake command line as well, and skip the foreach loop if the value is already set?

That would give the user full control, and allow wrappers around the Zephyr toolchain to put the cache somewhere more appropriate in special cases.

Reply:
Done